src/Questions.js

```diff
@@ -0,0 +1,130 @@
+import React, { useEffect, useState } from "react";
+const Question = (props) => {
+  const allAnswer = [
+    ...props.value.incorrect_answers,
+    props.value.correct_answer,
+  ];
+  const [selected, setSelected] = React.useState(null);
+  let styleQue = props.checkAns
+    ? selected === props.value.correct_answer
+      ? "bg-green-900  text-white border-none"
+      : "bg-red-900  text-white border-none opacity-50"
+    : "bg-violet-200 text-slate-700 border-none";
+
+  return (
+    <div className=" border-b py-2">
+      <h2
+        className="text-xl font-semibold text-slate-700"
+        dangerouslySetInnerHTML={{ __html: props.value.question }}
+      ></h2>
+      <div className="flex gap-2 lg:flex-nowrap flex-wrap my-3">
+        {allAnswer.sort().map((v, k) => {
+          return (
+            <div
+              key={k}
+              onClick={() => {
+                !props.checkAns && setSelected(v);
+                props.setAllClientAnswer((prev) => {
+                  prev[props.id] = {
+                    ...prev[props.id],
+                    isCollect: v === props.value.correct_answer ? true : false,
+                  };
+                  return prev;
+                });
+              }}
+              className={`rounded-xl border-2 py-1 px-5 cursor-pointer ${
+                v === selected && styleQue
+              }
+              ${
+                props.checkAns && props.value.correct_answer === v
+                  ? "bg-green-900 text-white border-none"
+                  : "border-slate-700"
+              }
+              `}
+            >
+              {v}
+            </div>
+          );
+        })}
+      </div>
+    </div>
+  );
+};
+export default function Questions() {
+  const [data, setData] = useState([]);
+  const [countCorrect, setCountCorrect] = useState(0);
+  const [checkAns, setCheckAns] = useState(false);
+  const [allClientAnswer, setAllClientAnswer] = useState(data);
```

Comment:
You should improve the naming by pluralizing `allClientAnswer` to become `allClientAnswers`